### ISSUE TITLE
branch destination out of range

### ISSUE THREAD

**Issue by JordanMajd** [huguesjohnson/RetailClerk89#1]
Building ROM with VASM on both my Linux and Windows machines are producing `error 2029: branch destination out of range`

```
./build.sh

... # java output nominal

Building ROM...
vasm 1.8e (c) in 2002-2018 Volker Barthelmann
vasm M68k/CPU32/ColdFire cpu backend 2.3c (c) 2002-2018 Frank Wille
vasm motorola syntax module 3.12 (c) 2002-2018 Frank Wille
vasm binary output module 1.8a (c) 2002-2009,2013,2015,2017 Volker Barthelmann

error 2029 in line 973 of "actionscript/ActionScriptDay04.X68": branch destination out of range
        included from line 43 of "actionscript/ActionScript.X68"
        included from line 140 of "RetailClerk89.X68"
>       beq.s   DialogTextCNowStaffDefaultTalk  ; not started, branch

error 2029 in line 975 of "actionscript/ActionScriptDay04.X68": branch destination out of range
        included from line 43 of "actionscript/ActionScript.X68"
        included from line 140 of "RetailClerk89.X68"
>       bne.s   DialogTextCNowStaffDefaultTalk  ; npc found, branch

error 2029 in line 1195 of "actionscript/ActionScriptDay04.X68": branch destination out of range
        included from line 43 of "actionscript/ActionScript.X68"
        included from line 140 of "RetailClerk89.X68"
>       beq.s   DialogTextFCBFishingStaffDefaultTalk    ; not started, branch

error 2029 in line 1197 of "actionscript/ActionScriptDay04.X68": branch destination out of range
        included from line 43 of "actionscript/ActionScript.X68"
        included from line 140 of "RetailClerk89.X68"
>       bne.s   DialogTextFCBFishingStaffDefaultTalk    ; npc found, branch
```

I'm guessing I missed a step somewhere along the way since I'm failing the same way on multiple systems. 

I used these flags to compile VASM:

```
# building on linux
make CPU=m68k SYNTAX=mot 

# building on windows.
md obj_win32
nmake /F Makefile.Win32 CPU=m68k SYNTAX=mot
```

I was able to build the ROM once by commenting out everything beyond day 4. However, the ROM just rendered flashing green to Gens.

I will spend some time tonight looking further into it.

**Comment by huguesjohnson** [comment]
It's a little counter-intuitive but you need to build vasm with Makefile.68k which should produce executables called vasmm68k_mot and vobjdump. On Linux those should be copied to /usr/local/bin.

**Comment by JordanMajd** [comment]
Thanks, I tried a little bit more last night, but still faced the same issue.

Compiling VASM with `make CPU=m68k SYNTAX=mot` uses default `Makefile` which produces both `vasmm68k_mot` and `vobjdump` for Unix. I've used these binaries to compile [basic templates](http://www.hacking-cult.org/?r/18/30) and other [example Genesis](https://github.com/sroccaserra/learning-megadrive) 68000 programs and run them on both Higan and Gens. 

According to [the docs](http://sun.hasenbraten.de/vasm/index.php?view=compile) `Makefile.68k` is used to cross-compile binaries to run on the Amiga. I tried building it just in case I overlooked or misunderstood something but wasn't able to produce a runnable binary. 

I'm not too worried about it, I'm working through your articles on developing Retail Clerk to learn the basics of Genesis dev. I was going to use this project to play around with the end result and as a reference. I can just download your ROM to see the end result, and the code still serves as a great reference.
